src/structs/pdb.rs

```diff
@@ -1017,8 +1017,49 @@ impl<'a> PDB {
     pub(crate) fn add_bond_counters(&mut self, atom1: usize, atom2: usize, bond: Bond) {
         self.bonds.push((atom1, atom2, bond));
     }
+
+    /// Returns a HashMap with the chains in contact within a given distance.
+    ///
+    /// # Arguments
+    ///
+    /// * `distance` - A f64 value representing the maximum distance between two atoms for them to be considered in contact.
+    ///
+    /// # Returns
+    ///
+    /// A HashMap with the chains in contact. The keys are the chain IDs and the values are vectors with the IDs of the chains in contact with the key chain.
+    pub fn chains_in_contact(&self, distance: f64) -> HashMap<String, Vec<String>> {
+        let mut chains = HashMap::new();
+        for chain1 in self.chains() {
+            for chain2 in self.chains() {
+                if chain1.id() == chain2.id() {
+                    continue;
+                }
+                for atom1 in chain1.atoms() {
+                    for atom2 in chain2.atoms() {
+                        if atom1.distance(atom2) < distance {
+                            chains
+                                .entry(chain1.id().to_owned())
+                                .or_insert_with(Vec::new)
+                                .push(chain2.id().to_owned());
+                            chains
+                                .entry(chain2.id().to_owned())
+                                .or_insert_with(Vec::new)
+                                .push(chain1.id().to_owned());
+                        }
```

Comment:
If you insert a `break` statement right after the insertion into the hashmap (before you close the if statement) you can short circuit the loop, prevent a lot of calculations from taking place and you will not need the deduplication code in lines 1054..=1057.

Reply:
Indeed the break reduces the calculations, good catch.

But the dedup is still needed since multiple `atom1` will have the same chain. The only way I can think of avoiding that is to check if it's already in the hashmap before adding - but I thought `or_insert_with` would do that... any suggestion?

Reply:
Could change to something like this to avoid the dedup -
```rust
let chain1_id = chain1.id().to_owned();
let chain2_id = chain2.id().to_owned();
let chain2_id_to_insert = chain2.id().to_owned();

let entry = chains.entry(chain1_id).or_insert_with(Vec::new);
if !entry.contains(&chain2_id) {
    entry.push(chain2_id_to_insert)
}
```

Reply:
Yes that would work, given that you do not rely of the sorted property of the output. One nit: I think you will be able to reuse `chain2_id` for both the lookup and insertion, but that you can try out when you actually write it.